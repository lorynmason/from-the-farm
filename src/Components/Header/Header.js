import React from 'react';

export const Header = () => {
  return (
    <header>
      <h1>from the farm</h1>
      <img id="logo" src="https://www.shareicon.net/data/2016/06/18/596437_pig_512x512.png" alt="from the farm cute pig logo"/>
      {/* <img id="logo" src="../../../public/assets/596437_pig_512x512.png" alt="from the farm cute pig logo"/> */}
    </header>
  )
}