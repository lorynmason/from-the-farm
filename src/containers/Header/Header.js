import React from 'react';
import { Link } from 'react-router-dom';
import image from '../../styles/images/pig-logo.png'
import { removeUser, isLoading, addMessage } from '../../actions';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

export const Header = (props) => {
  const clickBuy = () => {
    props.isLoading(true)
    setTimeout(() => {
      props.isLoading(false)
    }, 1000);
  }

  const signout = () => {
    props.removeUser();
    props.addMessage('You are now signed-out');
  }

  return (
    <header>
      <div className="header-logo">
      <Link to="/" onClick={clickBuy} title="Click to go to Home Page">
        <h1>from the farm</h1>
        <img id="logo" src={image} alt="from the farm cute pig logo"/>
      </Link>
      </div>
      <nav className="menu">
        <input type="checkbox" href="#" className="menu-open" name="menu-open" id="menu-open"/>
          <label className="menu-open-button" htmlFor="menu-open">
            <span className="hamburger hamburger-1"></span>
            <span className="hamburger hamburger-2"></span>
            <span className="hamburger hamburger-3"></span>
          </label>
          <Link to="/buy" onClick={clickBuy}><i className="fas fa-carrot"><p>buy</p></i></Link>
          <Link to="/profile"><i className="fas fa-user-circle"><p>my profile</p></i></Link>
          <Link to="/add-product"><i className="fas fa-plus-circle"><p>inventory</p></i></Link>
          <Link to="/" onClick={signout}><i className="fas fa-sign-out-alt"><p>sign out</p></i></Link>
      </nav>
    </header>
  );
}

export const mapDispatchToProps = (dispatch) => ({
  removeUser: () => dispatch(removeUser()),
  addMessage: (message) => dispatch(addMessage(message)),
  isLoading: (bool) => dispatch(isLoading(bool))
});

export default connect(null, mapDispatchToProps)(Header);

Header.propTypes = {
  removeUser: PropTypes.func,
  addMessage: PropTypes.func,
  isLoading: PropTypes.func
}
