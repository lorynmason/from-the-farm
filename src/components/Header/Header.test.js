import React from 'react';
import { Header } from './Header';
import { shallow } from 'enzyme'

describe('header', () => {
  let mockFunc = jest.fn()
  it('should match snapshot', () => {
    let wrapper = shallow(<Header removeUser={mockFunc} addMessage={mockFunc} isLoading={mockFunc}/>)
    expect(wrapper).toMatchSnapshot()
  });
  
})
