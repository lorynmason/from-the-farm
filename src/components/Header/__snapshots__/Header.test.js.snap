// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
      >
        <div
          className="header-logo"
        >
          <h1>
            from
            <span>
              the
            </span>
            farm
          </h1>
          <img
            alt="from the farm cute pig logo"
            id="logo"
            src="pig-logo.png"
          />
          <p>
            menu
            <i
              className="fas fa-caret-right"
            />
          </p>
          <div
            className="menu"
          >
            <Link
              replace={false}
              to="/buy"
            >
              <i
                className="fas fa-carrot"
              />
            </Link>
            <Link
              replace={false}
              to="/profile"
            >
              <i
                className="fas fa-user-circle"
              />
            </Link>
            <Link
              replace={false}
              to="/add-product"
            >
              <i
                className="fas fa-plus-circle"
              />
               
            </Link>
            <Link
              replace={false}
              to="/"
            >
              <i
                className="fas fa-sign-out-alt"
              />
            </Link>
          </div>
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="header-logo"
        >
          <h1>
            from
            <span>
              the
            </span>
            farm
          </h1>
          <img
            alt="from the farm cute pig logo"
            id="logo"
            src="pig-logo.png"
          />
          <p>
            menu
            <i
              className="fas fa-caret-right"
            />
          </p>
          <div
            className="menu"
          >
            <Link
              replace={false}
              to="/buy"
            >
              <i
                className="fas fa-carrot"
              />
            </Link>
            <Link
              replace={false}
              to="/profile"
            >
              <i
                className="fas fa-user-circle"
              />
            </Link>
            <Link
              replace={false}
              to="/add-product"
            >
              <i
                className="fas fa-plus-circle"
              />
               
            </Link>
            <Link
              replace={false}
              to="/"
            >
              <i
                className="fas fa-sign-out-alt"
              />
            </Link>
          </div>
        </div>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              from
              <span>
                the
              </span>
              farm
            </h1>,
            <img
              alt="from the farm cute pig logo"
              id="logo"
              src="pig-logo.png"
            />,
            <p>
              menu
              <i
                className="fas fa-caret-right"
              />
            </p>,
            <div
              className="menu"
            >
              <Link
                replace={false}
                to="/buy"
              >
                <i
                  className="fas fa-carrot"
                />
              </Link>
              <Link
                replace={false}
                to="/profile"
              >
                <i
                  className="fas fa-user-circle"
                />
              </Link>
              <Link
                replace={false}
                to="/add-product"
              >
                <i
                  className="fas fa-plus-circle"
                />
                 
              </Link>
              <Link
                replace={false}
                to="/"
              >
                <i
                  className="fas fa-sign-out-alt"
                />
              </Link>
            </div>,
          ],
          "className": "header-logo",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "from",
                <span>
                  the
                </span>,
                "farm",
              ],
            },
            "ref": null,
            "rendered": Array [
              "from",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "the",
                },
                "ref": null,
                "rendered": "the",
                "type": "span",
              },
              "farm",
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "from the farm cute pig logo",
              "id": "logo",
              "src": "pig-logo.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "menu",
                <i
                  className="fas fa-caret-right"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "menu",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-caret-right",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  replace={false}
                  to="/buy"
                >
                  <i
                    className="fas fa-carrot"
                  />
                </Link>,
                <Link
                  replace={false}
                  to="/profile"
                >
                  <i
                    className="fas fa-user-circle"
                  />
                </Link>,
                <Link
                  replace={false}
                  to="/add-product"
                >
                  <i
                    className="fas fa-plus-circle"
                  />
                   
                </Link>,
                <Link
                  replace={false}
                  to="/"
                >
                  <i
                    className="fas fa-sign-out-alt"
                  />
                </Link>,
              ],
              "className": "menu",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <i
                    className="fas fa-carrot"
                  />,
                  "replace": false,
                  "to": "/buy",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-carrot",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <i
                    className="fas fa-user-circle"
                  />,
                  "replace": false,
                  "to": "/profile",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-user-circle",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <i
                      className="fas fa-plus-circle"
                    />,
                    " ",
                  ],
                  "replace": false,
                  "to": "/add-product",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fas fa-plus-circle",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  " ",
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <i
                    className="fas fa-sign-out-alt"
                  />,
                  "replace": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-sign-out-alt",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <div
            className="header-logo"
          >
            <h1>
              from
              <span>
                the
              </span>
              farm
            </h1>
            <img
              alt="from the farm cute pig logo"
              id="logo"
              src="pig-logo.png"
            />
            <p>
              menu
              <i
                className="fas fa-caret-right"
              />
            </p>
            <div
              className="menu"
            >
              <Link
                replace={false}
                to="/buy"
              >
                <i
                  className="fas fa-carrot"
                />
              </Link>
              <Link
                replace={false}
                to="/profile"
              >
                <i
                  className="fas fa-user-circle"
                />
              </Link>
              <Link
                replace={false}
                to="/add-product"
              >
                <i
                  className="fas fa-plus-circle"
                />
                 
              </Link>
              <Link
                replace={false}
                to="/"
              >
                <i
                  className="fas fa-sign-out-alt"
                />
              </Link>
            </div>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="header-logo"
          >
            <h1>
              from
              <span>
                the
              </span>
              farm
            </h1>
            <img
              alt="from the farm cute pig logo"
              id="logo"
              src="pig-logo.png"
            />
            <p>
              menu
              <i
                className="fas fa-caret-right"
              />
            </p>
            <div
              className="menu"
            >
              <Link
                replace={false}
                to="/buy"
              >
                <i
                  className="fas fa-carrot"
                />
              </Link>
              <Link
                replace={false}
                to="/profile"
              >
                <i
                  className="fas fa-user-circle"
                />
              </Link>
              <Link
                replace={false}
                to="/add-product"
              >
                <i
                  className="fas fa-plus-circle"
                />
                 
              </Link>
              <Link
                replace={false}
                to="/"
              >
                <i
                  className="fas fa-sign-out-alt"
                />
              </Link>
            </div>
          </div>,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                from
                <span>
                  the
                </span>
                farm
              </h1>,
              <img
                alt="from the farm cute pig logo"
                id="logo"
                src="pig-logo.png"
              />,
              <p>
                menu
                <i
                  className="fas fa-caret-right"
                />
              </p>,
              <div
                className="menu"
              >
                <Link
                  replace={false}
                  to="/buy"
                >
                  <i
                    className="fas fa-carrot"
                  />
                </Link>
                <Link
                  replace={false}
                  to="/profile"
                >
                  <i
                    className="fas fa-user-circle"
                  />
                </Link>
                <Link
                  replace={false}
                  to="/add-product"
                >
                  <i
                    className="fas fa-plus-circle"
                  />
                   
                </Link>
                <Link
                  replace={false}
                  to="/"
                >
                  <i
                    className="fas fa-sign-out-alt"
                  />
                </Link>
              </div>,
            ],
            "className": "header-logo",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "from",
                  <span>
                    the
                  </span>,
                  "farm",
                ],
              },
              "ref": null,
              "rendered": Array [
                "from",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "the",
                  },
                  "ref": null,
                  "rendered": "the",
                  "type": "span",
                },
                "farm",
              ],
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "from the farm cute pig logo",
                "id": "logo",
                "src": "pig-logo.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "menu",
                  <i
                    className="fas fa-caret-right"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                "menu",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-caret-right",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Link
                    replace={false}
                    to="/buy"
                  >
                    <i
                      className="fas fa-carrot"
                    />
                  </Link>,
                  <Link
                    replace={false}
                    to="/profile"
                  >
                    <i
                      className="fas fa-user-circle"
                    />
                  </Link>,
                  <Link
                    replace={false}
                    to="/add-product"
                  >
                    <i
                      className="fas fa-plus-circle"
                    />
                     
                  </Link>,
                  <Link
                    replace={false}
                    to="/"
                  >
                    <i
                      className="fas fa-sign-out-alt"
                    />
                  </Link>,
                ],
                "className": "menu",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <i
                      className="fas fa-carrot"
                    />,
                    "replace": false,
                    "to": "/buy",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fas fa-carrot",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <i
                      className="fas fa-user-circle"
                    />,
                    "replace": false,
                    "to": "/profile",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fas fa-user-circle",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <i
                        className="fas fa-plus-circle"
                      />,
                      " ",
                    ],
                    "replace": false,
                    "to": "/add-product",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "fas fa-plus-circle",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "i",
                    },
                    " ",
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <i
                      className="fas fa-sign-out-alt"
                    />,
                    "replace": false,
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fas fa-sign-out-alt",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
